File: SocialAppFlutter/social_app/lib/services/events/generate_new_location_request.dart
class GenerateNewLocationRequest {
  final String city;
  final String locationName;
  final double latitude;
  final double longitude;
  final String ownerEmail;
  final int sportCategoryId;

  GenerateNewLocationRequest(this.city, this.locationName, this.latitude, this.longitude, this.ownerEmail, this.sportCategoryId);

  Map<String, dynamic> toJson() => 
  {
    'city': city,
    'locationName': locationName,
    'latitude': latitude,
    'longitude': longitude,
    'ownerEmail': ownerEmail,
    'sportCategoryId': sportCategoryId
  };
}

File: APIs/EventsService/EventsService/DTOs/Request/GenerateNewLocationRequest.cs
namespace EventsService.DTOs.Request
{
    public class GenerateNewLocationRequest
    {
        public string City { get; set; } = string.Empty;
        public string LocationName { get; set; } = string.Empty;
        public double Latitude { get; set; }
        public double Longitude { get; set; }
        public string OwnerEmail { get; set; } = string.Empty;
        public int SportCategoryId { get; set; }
    }
}
